### ISSUE TITLE
Incompatible method override not detected in for `__call__`

### ISSUE THREAD

**Issue by pierreglaser** [microsoft/pyright#7373]
Hi, and thanks for the great work on this tool.

Some incompatible overrides of the `__call__` method of a base class inside a subclass are not detected:

**Code or Screenshots**
Example script (`test.py`)

```python
class A:
    def __call__(self, x: int) -> int:
        return 1


class B(A):
    def __call__(self, y: str) -> int:
        return 1
```

Running `pyright 1.1.352` on it with the following `pyrightconfig.json`

```json
{
"reportIncompatibleMethodOverride": "error",
}
```

returns with no errors. In contrast, running `pyright 1.1.352` on 
```python
class A:
    def f(self, x: int) -> int:
        return 1


class B(A):
    def f(self, y: str) -> int:
        return 1
```
i.e the exact same script, but with the method `__call__` renamed to `f` returns the expected method override error

```
/private/tmp/repro/test.py
  /private/tmp/repro/test.py:7:9 - error: Method "f" overrides class "A" in an incompatible manner
    Parameter 2 name mismatch: base parameter is named "x", override parameter is named "y" (reportIncompatibleMethodOverride)
1 error, 0 warnings, 0 informations
```

I believe the correct behavior should be pyright flagging the incompatible method override in both cases.

Thanks!

**Comment by erictraut** [comment]
Thanks for the bug report.

Here's what's going on. Pyright doesn't enforce parameter name consistency for dundered method overrides. There are too many examples in typeshed where parameter names are inconsistent for dundered methods, so this exemption is required to avoid a bunch of noise (false positives). Mypy applies a similar exemption.

However, the code in pyright that was applying the exemption went too far and also prevented checks for parameter _type_ consistency. If you change your code sample to use the same parameter name `x` in both cases, you'll see that it detects the type mismatch.

This will be addressed in the next release.

**Comment by pierreglaser** [comment]
I see, thanks for the quick response.

> Here's what's going on. Pyright doesn't enforce parameter name consistency for dundered method overrides. There are too many examples in typeshed where parameter names are inconsistent for dundered methods, so this exemption is required to avoid a bunch of noise (false positives). 

Is this documented somewhere?

**Comment by erictraut** [comment]
This is included in pyright 1.1.353, which I just published.
